src/main/webapp/app/overview/course-statistics/course-statistics.component.ts

```diff
@@ -236,11 +237,12 @@ export class CourseStatisticsComponent implements OnInit, OnDestroy {
     }
 
     groupExercisesByType() {
-        const exercises = this.course!.exercises;
+        let exercises = this.course!.exercises;
         const groupedExercises: any[] = [];
         const exerciseTypes: string[] = [];
+        exercises = exercises.sort((a: Exercise, b: Exercise) => (a.dueDate || moment().add(5, 'year')).valueOf() - (b.dueDate || moment().add(5, 'year')).valueOf());
```

Comment:
I would prefer not to assign the result of `Array.Prototype.sort`, because it sorts in place.
So it's misleading to somebody who is not familiar with this method.

If there are no performance indications I would in general prefer lodash `sortBy` (https://lodash.com/docs/#sortBy).
```exercises = sortBy(exercises, (a, b)...)```

Reply:
i replaced the function